src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,333 @@
+    const aInt = parseInt(a, 10);
+    const bInt = parseInt(b, 10);
+    const testInt = parseInt(test, 10);
+    const min = aInt < bInt ? aInt : bInt;
+    const max = aInt > bInt ? aInt : bInt;
+    return (testInt >= min && testInt <= max);
+  }
+
+  /**
+   * Checks if integer representing port is valid.
+   * @param {number} port port number being checked.
+   * @return {boolean}
+   */
+  public static isValidIPv4AddressPort(port: number): boolean {
+    return (port >= 1 && port <= 65535);
+  }
+
+  /**
+   * Gets the number of IP addresses in a given range. Although unlikely to be
+   * an issue, note that due to the fact that javascript numbers are represented
+   * as signed, two's complement integers, for a network with more than 2^31
+   * addresses this function will return a negative number (May not even be
+   * technically possible). Returns -1 if the parameters are invalid.
+   * @param {IPv4AddressRange} ipRange object representing IP range of the form { start:
+   *          "x.x.x.x", end: "x.x.x.x" }
+   * @return {integer} size of range.
+   */
+  public static getNumberOfIPv4AddressesInIPv4AddressRange(range: IPv4AddressRange): number | null {
+    return this.getIPv4AsInteger(new IPv4Address(range.end)) - this.getIPv4AsInteger(new IPv4Address(range.start)) + 1;
+  }
+
+  /**
+   * Returns an available IP address within one of an array of given ranges,
+   * given an array of the IP addresses that are already allocated. Returns null
+   * if no addresses are unallocated.
+   * @param {IPv4AddressRange[]} ipRanges array of IP ranges to choose from.
+   * @param {string[]} allocatedIpAddresses array of IP addresses which have
+   *          already been allocated.
+   * @return {string} an available IP address.
+   */
+  public static getAvailableIPv4(ipv4AddressRanges: Array<IPv4AddressRange>,
+                                 allocatedIpAddresses: IPv4Address[]): string | null {
+    const allocInts = allocatedIpAddresses.map(
+      (ipAddress: IPv4Address) => {
+        return this.getIPv4AsInteger(ipAddress);
+      }
+    );
+    for (const ipIPv4AddressRange of ipv4AddressRanges) {
+      const start = this.getIPv4AsInteger(new IPv4Address(ipIPv4AddressRange.start));
+      const end = this.getIPv4AsInteger(new IPv4Address(ipIPv4AddressRange.end));
+      for (let j = start; j !== end; j++) {
+        if (!allocInts.includes(j)) {
+          return this.getIntegerAsIPv4Address(j);
+        }
+      }
+      if (!allocInts.includes(start) && start === end) {
+        return this.getIntegerAsIPv4Address(start);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns if every address within range is in ip list.
+   * @param {IPv4AddressRange} range range object.
+   * @param {string[]} ipList array of IP addresses.
+   * @return {boolean}
+   */
+  public static areAllIPv4AddressesFromIPv4AddressRangeInList(ipList: IPv4Address[], range: IPv4AddressRange): boolean {
```

Comment:
probably don't need to repeat IPv4Address twice in the method name, I'd remove the second occurrence for the sake of brevity.